spike.rb

```diff
@@ -0,0 +1,40 @@
+require 'gosu'
+require 'chipmunk'
+require_relative 'chip-gosu-functions'
+require_relative 'obj'
+
+class Spike < Obj
+  def initialize(window, g, sizex, sizey, angle = 0)
+    super window, g, "resources/images/spikes.png"
+
+    @sizex = sizex
+    @sizey = sizey
+
+    @body = CP::Body.new Float::INFINITY, Float::INFINITY
+    @shapes << CP::Shape::Poly.new(body, [CP::Vec2.new(0.0, @sizex / 8.0), CP::Vec2.new(- @sizey, 0.0), CP::Vec2.new(- @sizey  , 2 * @sizex / 8.0)], CP::Vec2.new(0,0))
```

Comment:
Quite a lot of repetition here. Maybe create a method for this? Also, split this into multiple lines and remove useless parentheses.